### PR TITLE
Exposes resolvers configurations

### PR DESCRIPTION
**Pull request by sm43**
# Changes
this allows user to enable/disable resolvers through TektonConfig and provide
configurations for each of the resolver.
to enable/disable resolvers, following are the fields
```
    enable-bundles-resolver: true
    enable-cluster-resolver: true
    enable-git-resolver: true
    enable-hub-resolver: true
```
to provide configurations for individual providers, following fields are defined
which are map[string]string
and after defining **only valid fields will be copied and invalid fields will be ignored.**
NOTE: we currently **_don't_** **validate** or **default** configurations for individual resolvers
```
     bundles-resolver-config:
        default-service-account: abc
      cluster-resolver-config:
        default-namespace: test
      git-resolver-config:
        server-url: localhost.com
      hub-resolver-config:
        default-tekton-hub-catalog: cat
```

the Tektonconfig would look like below
```
apiVersion: operator.tekton.dev/v1alpha1
kind: TektonConfig
metadata:
  name: config
spec:
  pipeline:
    bundles-resolver-config:
      default-service-account: abc
    cluster-resolver-config:
      default-namespace: test
    disable-creds-init: false
    embedded-status: full
    enable-api-fields: stable
    enable-bundles-resolver: true
    enable-cluster-resolver: true
    enable-custom-tasks: false
    enable-git-resolver: true
    enable-hub-resolver: true
    enable-tekton-oci-bundles: false
    git-resolver-config:
      server-url: localhost.com
    hub-resolver-config:
      default-tekton-hub-catalog: cat
    metrics.pipelinerun.duration-type: histogram
    metrics.pipelinerun.level: pipeline
    metrics.taskrun.duration-type: histogram
    metrics.taskrun.level: task
    require-git-ssh-secret-known-hosts: false
    running-in-environment-with-injected-sidecars: true
    send-cloudevents-for-runs: false
```

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Run `make test lint` before submitting a PR
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

**Review by savitaashture — APPROVED** [review]

File: test/e2e/kubernetes/tektonresultdeployment_test.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ import (
 
 // TestTektonResultDeployment verifies the TektonResult creation, deployment recreation, and TektonResult deletion.
 func TestTektonResultDeployment(t *testing.T) {
+	t.Skip()
```
Comment (sm43) [inline test/e2e/kubernetes/tektonresultdeployment_test.go:40]: https://github.com/tektoncd/operator/issues/1148
need to check in results

**Review by vdemeester — COMMENTED** [review]
I wonder if we could have done something like:

```
resolvers:
  git:
    enable: true
    server-url: https://…
  hub:
    enable: true
    …
```

But I think we can think about this for a v1beta1 version of the operator CRD or something.

File: pkg/apis/operator/v1alpha1/tektonpipeline_defaults.go [hunk 2]
```diff
@@ -80,6 +80,20 @@ func (p *Pipeline) setDefaults() {
 		p.MetricsTaskrunLevel = config.DefaultTaskrunLevel
 	}
 
+	// Resolvers
```
Comment (piyush-garg) [inline pkg/apis/operator/v1alpha1/tektonpipeline_defaults.go:83]: are we enabling by default?

Comment (sm43) [reply pkg/apis/operator/v1alpha1/tektonpipeline_defaults.go:83]: yes

Comment (vdemeester) [reply pkg/apis/operator/v1alpha1/tektonpipeline_defaults.go:83]: yes starting in 0.41 they are enabled by default.

**Comment by sm43** [comment]
> One suggestion
> 
> instead of
> 
> ```
>     bundles-resolver-config:
>       default-service-account: abc
>     enable-bundles-resolver: true
> ```
> 
> we should do
> 
> ```
>     bundles-resolver:
>         enable: true
>         default-service-account: abc
> ```
> 
> or something similar, I feel the config of the resolver should be nested there only, not in a separate field



> 

@piyush-garg 
yeah it make sense but it wouldn't work with existing transformers directly 😄 
should we try that for v1beta1 api?

**Comment by piyush-garg** [comment]
> > One suggestion
> > instead of
> > ```
> >     bundles-resolver-config:
> >       default-service-account: abc
> >     enable-bundles-resolver: true
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > we should do
> > ```
> >     bundles-resolver:
> >         enable: true
> >         default-service-account: abc
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > or something similar, I feel the config of the resolver should be nested there only, not in a separate field
> 
> > 
> 
> @piyush-garg yeah it make sense but it wouldn't work with existing transformers directly smile should we try that for v1beta1 api?

okay, lets target for v1beta1 then

/lgtm

**Pull request merged**